### ISSUE TITLE
[react-native] Error: stream could not be found

### ISSUE THREAD

**Issue by doronpr** [typeorm/typeorm#7976]
## Issue Description

In react-native we started to get this error:

`node_modules/typeorm/browser/driver/sqlserver/SqlServerQueryRunner.js: stream could not be found`

This is due to this change: https://github.com/typeorm/typeorm/blame/master/src/driver/sqlserver/SqlServerQueryRunner.ts#L317
`stream` is part of node and is not available in `react-native`.

*Regression in 0.2.3.5* FYI @imnotjames 

### My Environment
Any `react-native` version.

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Mac OS         |
| Node.js version     | 14.17.3  | 

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✅  No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by imnotjames** [comment]
Are you getting this as part of the build?

**Comment by imnotjames** [comment]
Do you have an example repository that would exhibit this so I can easily confirm the problem and any fixes?

**Comment by doronpr** [comment]
@imnotjames I prepared min reproduce script:

```
npx create-react-native-app min1
cd ./min1
yarn add typeorm
echo "require('typeorm');" >> ./index.js
react-native bundle --entry-file ./index.js --bundle-output foo
```

**Comment by imnotjames** [comment]
It seems a longstanding issue was that we didn't correctly set the `browser` field.

It had `.ts` instead of `.js` in some cases which caused issues.

That's been fixed with #7982 & I'll probably cut a new release this weekend with this fix and few others.

**Comment by imnotjames** [comment]
To test, I used the following commands:

```
npx create-react-native-app react-native-typeorm-example
cd ./react-native-typeorm-example
yarn add typeorm@0.2.36-dev.0d90bcd
echo "require('typeorm');" >> ./index.js
npx react-native bundle --entry-file ./index.js --bundle-output foo
```
